src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -797,44 +800,76 @@ private function registerValidationConfiguration(array $config, ContainerBuilder
-            $files[0][] = dirname($reflClass->getFileName()).'/Resources/config/validation.xml';
-            $container->addResource(new FileResource($files[0][0]));
+            $files['xml'][] = $file = dirname($reflClass->getFileName()).'/Resources/config/validation.xml';
+            $container->addResource(new FileResource($file));
         }
 
         $bundles = $container->getParameter('kernel.bundles');
         foreach ($bundles as $bundle) {
             $reflection = new \ReflectionClass($bundle);
             $dirname = dirname($reflection->getFileName());
 
-            if (is_file($file = $dirname.'/Resources/config/validation.xml')) {
-                $files[0][] = realpath($file);
+            if (is_file($file = $dirname.'/Resources/config/validation.yml')) {
+                $files['yml'][] = realpath($file);
                 $container->addResource(new FileResource($file));
             }
 
-            if (is_file($file = $dirname.'/Resources/config/validation.yml')) {
-                $files[1][] = realpath($file);
+            if (is_file($file = $dirname.'/Resources/config/validation.xml')) {
+                $files['xml'][] = realpath($file);
                 $container->addResource(new FileResource($file));
             }
 
             if (is_dir($dir = $dirname.'/Resources/config/validation')) {
-                foreach (Finder::create()->files()->in($dir)->name('*.xml') as $file) {
-                    $files[0][] = $file->getRealpath();
-                }
-                foreach (Finder::create()->files()->in($dir)->name('*.yml') as $file) {
-                    $files[1][] = $file->getRealpath();
-                }
-
+                $this->getValidatorMappingFilesFromDir($dir, $files);
                 $container->addResource(new DirectoryResource($dir));
             }
         }
+    }
 
-        return $files;
+    /**
+     * Search for validation files in directory.
+     *
+     * @param string $dir   The directory that contains the validation files
+     * @param array  $files A array reference of validation files
+     */
+    private function getValidatorMappingFilesFromDir($dir, array &$files)
+    {
+        foreach (Finder::create()->files()->in($dir)->name('/\.(xml|ya?ml)$/') as $file) {
+            $files[$file->getExtension()][] = $file->getRealpath();
+        }
+    }
+
+    /**
+     * Get all validation files from config.
+     *
+     * @param array $config A validation configuration array
+     * @param array $files  A array reference of validation files
+     */
+    private function getValidatorMappingFilesFromConfig(array $config, array &$files)
+    {
+        foreach ($config['mapping']['paths'] as $path) {
+            if (is_dir($path)) {
+                $this->getValidatorMappingFilesFromDir($path, $files);
+            } elseif (is_file($path)) {
+                if (preg_match('/\.(xml|ya?ml)$/', $path, $matches)) {
+                    $files[$matches[0]][] = realpath($path);
```

Comment:
`$matches[0]` will contain the string `yaml` when the file extension is `.yaml` which would now be ignored.